pkg/reconciler/managed/reconciler.go

```diff
@@ -675,14 +764,21 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"external-name", meta.GetExternalName(managed),
 	)
 
+	managementPoliciesEnabled := r.features.Enabled(feature.EnableAlphaManagementPolicies)
+	if managementPoliciesEnabled {
+		log.WithValues("managementPolicies", managed.GetManagementPolicy())
+	}
+	actionChecker := NewManagementPolicyActionChecker(managementPoliciesEnabled, managed.GetManagementPolicy())
```

Comment:
Would there be any benefit to injecting this as a dependency (i.e. making it a field of `Reconciler`? In practice I doubt it would ever be swapped out in production, but maybe it would help to be able to swap it out in tests?

Reply:
I aimed for that initially, but it needs a management policy, which we get from the managed resource we GET somewhere at the beginning of the reconcile loop. So we could inject it without the policy and then pass it later as an method argument,  or inject it without and set it later, but that seemed like not the best solution to me.

Maybe there is another way that I am not seeing.